File: server/serverpackets/S_ServerVersion.py
# -*- coding: utf-8 -*-

import time
from Config import Config
from ..codes import Opcodes
from ServerBasePacket import ServerBasePacket

class S_ServerVersion(ServerBasePacket):
    def __init__(self):
        ServerBasePacket.__init__(self)
        self.writeC(Opcodes.S_OPCODE_SERVERVERSION)
        self.writeC(0x00)
        self.writeC(0x02)
        self.writeD(0x00a8c732)  # server verion 3.5C Taiwan Server
        self.writeD(0x00a8c6a7)  # cache verion 3.5C Taiwan Server
        self.writeD(0x77cf6eba)  # auth verion 3.5C Taiwan Server
        self.writeD(0x00a8cdad)  # npc verion 3.5C Taiwan Server
        self.writeD(int(time.time()))
        self.writeC(0x00)
        self.writeC(0x00)
        self.writeC(Config.getint('server', 'ClientLanguage'))  # Country: 0.US 3.Taiwan 4.Janpan 5.China
        self.writeD(0x00000000)
        self.writeC(0xae)
        self.writeC(0xb2)

File: ServerPackets/S_ServerVersion.cpp
//
// Created by 范炜东 on 2019/1/16.
//

#include <Poco/Timestamp.h>
#include <Poco/Util/Application.h>
#include "Opcodes.h"
#include "S_ServerVersion.h"

namespace Lineage {

using Poco::Timestamp;
using Poco::Util::Application;

S_ServerVersion::S_ServerVersion() : ServerPacket() {
    Application &app = Application::instance();

    writeC(S_OPCODE_SERVERVERSION);
    writeC(0x00);
    writeC(0x02);
    writeD(0x00a8c732);
    writeD(0x00a8c6a7);
    writeD(0x77cf6eba);
    writeD(0x00a8cdad);
    Timestamp t;
    writeD(t.epochMicroseconds() / 1000);
    writeC(0x00);
    writeC(0x00);
    writeC(app.config().getInt("Server.language", 5));
    writeD(0x00000000);
    writeC(0xae);
    writeC(0xb2);
}

}
